Stores grid screen

ilijaS opened the pull request:
- created UI for displaying stores in grid
- created network communication module; both mock and real data

ilijaS committed:
Stores grid screen

-Design and develop model for app
-Create network communication module
-Create network communication mock module
-Create UI for displaying stores in grid

ilijaS committed:
Stores grid screen

-Created UI for displaying the stores in grid
-Created network communication module - both mock and real data

ilijaS committed:
commit for review

marko-lf reviewed (changes requested):
LocationModule:
    var distanceDict:[Int16:String] 

Location shouldn't know about Store at all. StoreModule can use Location to calculate it's distance. Also, putting those calculations in getter is kinda heavy. Distance data should be reused with option to refresh.

Network module shouldn't know about Store either.
Imagine you have multiple models using network module and same method to load their data. It's silly that network updates some model that didn't ask for it.
Solution:
Use completion blocks to retrieve data from network module and parse them into Store objects.
Note: eliminate dependency for storeModel in network module. 

protocols (store, location, network)
Their methods shouldn't be imperative: reload, show, enable, disable, etc. They should inform that some event occured: e.g didLoadData, didFailLoading, storeDataValidated, etc.
Also, enableUserInteraction shouldn't be part of protocol. VC should know when to enable and when to disable user interaction. Example: you have one VC and multiple models. One model tells VC to disableUserInteraction and other says to enable.
How to solve: Question model for data, if the data is not ready block UI until data is ready (until model says data is ready via delegation)

Storyboard has no constraints. We neeeeeed constraints :)

ilijaS committed:
Code review redoing 1/2

-added constraints to elements of the cell
- removed unwanted dependencies, redone the data fetch logic, removed distanceDict from locationmanager logic

TODO:
-redo protocol methods

ilijaS committed:
git ignore redone

removed the dotless gitignore file
rewritten the .gitignore file

ilijaS committed:
more review requirement tweaks

protocol functionalities tweaked

marko-lf commented:
There are some errors in storyboard. I think we don't need stack for lay-outing our labels.
Labels are somehow transparent, maybe because they are children of a transparent view? 😄

ilijaS committed:
Labels in cells transparency issue fixed

-labels no longer transparent
-delegate method definitions moved to extnesions

marko-lf commented:
👍 🥇
